dropshot/tests/common/mod.rs

```diff
@@ -51,87 +51,84 @@ pub fn create_log_context(test_name: &str) -> LogContext {
-        );
-        let end_cert = rustls::pki_types::CertificateDer::from(
-            end_keypair
-                .serialize_der_with_signer(&intermediate_keypair)
-                .expect("failed to serialize end-entity cert"),
-        );
+        let intermediate_cert = intermediate_params
+            .signed_by(&intermediate_keypair, &root_cert, &root_keypair)
+            .expect("failed to sign intermediate cert");
+
+        let end_keypair =
+            rcgen::KeyPair::generate().expect("end keypair generation failed");
+        let end_params =
+            rcgen::CertificateParams::new(vec!["localhost".into()])
+                .expect("invalid end params");
+        let end_cert = end_params
+            .signed_by(&end_keypair, &intermediate_cert, &intermediate_keypair)
+            .expect("failed to sign end cert");
 
         Self {
             root_cert,
-            intermediate_cert,
             intermediate_keypair,
-            end_cert,
+            intermediate_cert,
             end_keypair,
+            end_cert,
         }
     }
 
-    pub fn end_cert_private_key<'b>(
-        &self,
-    ) -> rustls::pki_types::PrivateKeyDer<'b> {
+    pub fn end_cert_private_key(&self) -> rustls::pki_types::PrivateKeyDer {
         rustls::pki_types::PrivateKeyDer::from(
-            rustls::pki_types::PrivatePkcs1KeyDer::from(
-                self.end_keypair.serialize_private_key_der(),
+            rustls::pki_types::PrivatePkcs8KeyDer::from(
+                self.end_keypair.serialize_der(),
             ),
         )
     }
 
     pub fn cert_chain(&self) -> Vec<rustls::pki_types::CertificateDer> {
         vec![
-            self.end_cert.clone(),
-            self.intermediate_cert.clone(),
-            self.root_cert.clone(),
+            self.end_cert.der().clone(),
+            self.intermediate_cert.der().clone(),
+            self.root_cert.der().clone(),
         ]
     }
 
     pub fn generate_new_end_cert(&mut self) {
-        let end_keypair = rcgen::Certificate::from_params(
-            rcgen::CertificateParams::new(vec!["localhost".into()]),
-        )
-        .expect("failed to generate end-entity keys");
-        let end_cert = rustls::pki_types::CertificateDer::from(
-            end_keypair
-                .serialize_der_with_signer(&self.intermediate_keypair)
-                .expect("failed to serialize end-entity cert"),
-        );
+        let end_keypair =
+            rcgen::KeyPair::generate().expect("end keypair generation failed");
+        let end_params =
+            rcgen::CertificateParams::new(vec!["localhost".into()])
+                .expect("invalid end params");
+        let end_cert = end_params
+            .signed_by(
+                &end_keypair,
+                &self.intermediate_cert,
+                &self.intermediate_keypair,
+            )
+            .expect("failed to sign end cert");
         self.end_keypair = end_keypair;
         self.end_cert = end_cert;
     }
```

Comment:
```suggestion
```

It looks like this function is unused.

Side note: GitHub got very confused by this suggestion.  I'm just trying to suggest ripping this function out.